apis/duck/v1/source_types.go

```diff
@@ -170,3 +171,50 @@ type SourceList struct {
 
 	Items []Source `json:"items"`
 }
+
+func (s *Source) Validate(ctx context.Context) *apis.FieldError {
+	if s == nil {
+		return nil
+	}
+	return ValidateSource(ctx, *s).ViaField(apis.CurrentField)
```

Comment:
`ValidateSource` is a useless method, just call `src.Spec.Validate(ctx).ViaField("spec")`directly.